server/app/auth/oidc/CiviformOidcLogoutActionBuilder.java

```diff
@@ -0,0 +1,98 @@
+package auth.oidc;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import auth.CiviFormProfileData;
+import com.google.common.collect.ImmutableMap;
+import com.nimbusds.openid.connect.sdk.LogoutRequest;
+import com.typesafe.config.Config;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.context.session.SessionStore;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.exception.http.RedirectionAction;
+import org.pac4j.core.profile.UserProfile;
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.core.util.HttpActionHelper;
+import org.pac4j.oidc.config.OidcConfiguration;
+import org.pac4j.oidc.logout.OidcLogoutActionBuilder;
+
+/**
+ * Custom OidcLogoutActionBuilder for CiviFormProfileData (since it extends CommonProfile, not
+ * OidcProfile). Also allows for divergence from the [oidc
+ * spec](https://openid.net/specs/openid-connect-rpinitiated-1_0.html) if your provider requires it
+ * (e.g. Auth0).
+ *
+ * <p>Does not provide the recommended id_token_hint, since these are not stored by the civiform
+ * profile.
```

Comment:
question (non-blocking): should we store this?

Reply:
I don't think it's worth storing this until one the civic entities' identity providers requires it in order to logout correctly.